drivers/sensors/spacemouse_module.c

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 Christopher Courtney, aka Drashna Jael're  (@drashna) <drashna@live.com>
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "spacemouse_module.h"
+#include "pointing_device_internal.h"
+#include "uart.h"
+
+// datasheet available at https://3dconnexion.com/cn/wp-content/uploads/2020/02/HW-Spec-3DX-700039_Rev001_serial.pdf
+
+#define SPACEMOUSE_INPUT_OFFSET (8192)
+
+bool spacemouse_send_command(uint8_t cmd) {
+    uart_write(cmd);
+    uint8_t buf[2];
+    uart_receive(buf, 2);
+    return (buf[0] == cmd && buf[1] == SPACEMOUSE_CMD_END);
+}
+
+/**
+ * @brief Set the zero position of the module
+ *
+ * @return true command ran successfully
+ * @return false command failed
+ */
+bool spacemouse_cmd_set_zero_position(void) {
+    return spacemouse_send_command(SPACEMOUSE_CMD_SET_ZERO_POSITION);
+}
+
+/**
+ * @brief Starts automatic transmission of data, at 30ms invervals
+ * Automatic data transmission happens at 30 ms intervals, but device can be polled at 100/s or 10ms intervals
```

Comment:
Did the 10ms interval come from testing? I couldn't see it in the datasheet.

Reply:
5.1.1, for the uart settings.  100/s for the data rate (page 10)